crates/bevy_ecs/src/system/query.rs

```diff
@@ -11,102 +11,212 @@ use bevy_tasks::TaskPool;
+/// 2. **Query filters (optional):** a predicate that ignores query results that don't match
+///    its conditions.
 ///
-/// The main way to access the components of a query is through the [`Query::iter`] and [`Query::iter_mut`]
-/// functions which return a [`QueryIter`] to iterate over:
+/// # Usage as system parameter
+///
+/// A query is defined by declaring it as a system parameter. This section shows the various
+/// use cases of `Query` as a system parameter.
+///
+/// ## Immutable component access
+///
+/// The following example defines a query that gives an iterator over `(&ComponentA, &ComponentB)`
+/// tuples, where `ComponentA` and `ComponentB` belong to the same entity. Accessing components
+/// immutably helps system parallelization.
 ///
 /// ```
 /// # use bevy_ecs::system::IntoSystem;
 /// # use bevy_ecs::system::Query;
-/// struct UnitHealth(pub u32);
-/// fn system(query: Query<&UnitHealth>) {
-///     for UnitHealth(health) in query.iter() {
-///         println!("We got {} health points left!", health);
-///     }
-/// }
+/// # struct ComponentA;
+/// # struct ComponentB;
+/// # fn system(
+/// query: Query<(&ComponentA, &ComponentB)>
+/// # ) {}
 /// # system.system();
 /// ```
 ///
-/// ### Multiple Component Access
+/// ## Mutable component access
 ///
-/// Instead of asking for just one component like before we can build a query that queries for multiple
-/// components with the help of tuples,`Query<(&Shape, &Color, &mut Size)>`. This query retrieves
-/// immutable references to the `Shape` and `Color` component and a mutable reference to the `Size`
-/// component.
+/// The following example is similar to the previous one, with the exception of `ComponentA`
+/// being accessed mutably here. Note that both mutable and immutable accesses are allowed
+/// in the same query.
 ///
 /// ```
 /// # use bevy_ecs::system::IntoSystem;
 /// # use bevy_ecs::system::Query;
-/// #[derive(Debug)]
-/// enum Shape {
-///     Circle,
-///     Box,
-/// };
-/// struct Color(pub String);
-/// struct Size(pub u32);
-/// fn system(mut query: Query<(&Shape, &Color, &mut Size)>) {
-///     for (shape, color, mut size) in query.iter_mut() {
-///         *size = Size(1);
-///         println!("We got a {} colored {:?} and made it one unit big!", color.0, shape);
-///     }
-/// }
+/// # struct ComponentA;
+/// # struct ComponentB;
+/// # fn system(
+/// // `ComponentA` is accessed mutably, while `ComponentB` is accessed immutably.
+/// mut query: Query<(&mut ComponentA, &ComponentB)>
+/// # ) {}
 /// # system.system();
 /// ```
 ///
-/// Note the use of [`Query::iter_mut`] here, as our query is not read-only anymore due to the use
-/// of the `&mut` [`WorldQuery`] we aren't able to use the `iter` method any longer.
+/// Two systems cannot be executed in parallel if both access a certain component and
+/// at least one of the accesses is mutable, unless the schedule can verify that no entity
+/// could be found in both queries.
 ///
```

Comment:
```suggestion
/// Two systems cannot be executed in parallel if both access a certain component and
/// at least one of the accesses is mutable, unless the schedule can verify that no entity
/// could be found in both queries, as otherwise Rusts mutability Rules would be broken.
///
/// Similialry a System cannot contain two Queries that would break Rusts mutability Rules.
/// If you need such Queries, you can use Filters to make the Queries disjoint or use a [`QuerySet`](super::QuerySet).
```

Reply:
Typo in "similarly" in the suggested change.